### ISSUE TITLE
Migrate from cra-env to import-meta-env alongside the migration from CRA to Vite

### ISSUE THREAD

**Issue by olevitt** [InseeFrLab/onyxia#632]
Hi ! 

Currently the web image uses a sophisticated entrypoint to translate env variables into a configuration file at runtime. This results in the need of additional software (currently npm) on top of nginx and a resulting image currently weighting 250M.
We could get rid of this env variable translation logic in favor of a simple file configuration which would be in most cases done using configmaps. 
This would be a breaking change requiring documentation changes as well as some rewriting on the helm chart (adding and mouting a configmap) but would result in a cleaner and smaller (by a huge margin).

**Comment by garronej** [comment]
Strongly opposed to this idea.
I've developed a specialized project for bundling runtime environments, which is thoroughly tested, well-documented, and offers an excellent development experience.
https://github.com/garronej/cra-envs
The type definitions are auto-generated from their declarations, which is a significant advantage, also it's iso with how CRA handle environement variables by default.  
Bundling Node may not be ideal, but it has virtually no impact on performance or functionality.

I would, however, be open to migrating to [import-meta-env](https://iendeavor.github.io/import-meta-env/) once we switch to Vite.
This move would introduce no breaking changes. Additionally, unlike cra-env, import-meta-env generates a shell script, eliminating the need to bundle Node while retaining the avantages in terms of DX.

**Comment by garronej** [comment]
Not mentioning the fact that the index.html is a esm template that get rendered at container startup into html using the configuration.  
The suggered approach woudn't cover this usecase.

**Comment by olevitt** [comment]
> I would, however, be open to migrating to [import-meta-env](https://iendeavor.github.io/import-meta-env/) once we switch to Vite. This move would introduce no breaking changes. Additionally, unlike cra-env, import-meta-env generates a shell script, eliminating the need to bundle Node while retaining the avantages in terms of DX.

After reading some documentation on `import-meta-env` and taking a look at their examples, I would be okay with something like this : https://github.com/iendeavor/import-meta-env/blob/main/packages/examples/docker-starter-example/Dockerfile 
As long as we have a plain vanilla nginx as runtime I'm okay with whatever best suits your needs :)

**Comment by garronej** [comment]
Ok De

**Comment by garronej** [comment]
Ok deal for migrating to import-meta-env when we migrate to Vite.

**Comment by garronej** [comment]
Closing
Onyxia is designed to be customizable without requiring a rebuild. This allows administrators to configure their own font, page description, social media previews, and other elements that need to be present in the HTML head.
Using cra-envs enables us to re-render the index EJS template upon container startup this is something something that, `import-meta-env` will never do.
We've determined that including Node.js in the bundle only adds a 58MB overhead, which is negligible given what it enables.
The roadmap for migrating to Vite simply involves adding Vite support to [cra-envs](https://github.com/garronej/cra-envs), and possibly renaming the library in the process.  

Related to #570
